ibis/clickhouse/compiler.py

```diff
@@ -101,6 +101,32 @@ class ClickhouseTableSetFormatter(comp.TableSetFormatter):
         ops.AnyLeftJoin: 'ANY LEFT JOIN'
     }
 
+    def _walk_join_tree(self, op):
+        left = op.left.op()
+        right = op.right.op()
+
+        if util.all_of([left, right], ops.Join):
+            raise NotImplementedError('Do not support joins between '
+                                      'joins yet')
+
+        self._validate_join_predicates(op.predicates)
+
+        join_name = self._get_join_type(op)
+
+        # Both tables
+        left, right = op.left, op.right
+
+        if not left._is_materialized():
+            left = left.materialize()
+
+        if not right._is_materialized():
+            right = right.materialize()
```

Comment:
Should this call `_materialize_if_necessary`?

Reply:
Not exactly. In `_get_schema` we need to materialize unconditionally, whereas `_materialize_if_necessary` checks for string predicates (column names).